### PR TITLE
Feature/mailer

### PR DESCRIPTION
**Pull request by str1nker**
Added mailer module to Squadster

Configured with Mailgun

To make it work, you need to set ENV variables. I will send values directly
`export MAILGUN_API_KEY`
`export MAILGUN_DOMAIN`
`export DEFAULT_FROM_EMAIL`

Also, we have a test restricted account, so we can send mails to only 5 different people.
To be able to send mails to other people we need to add custom DNS, I will share accounts creds for that if necessary.

### CODE REVIEW

File: lib/squadster/accounts/tasks/notify.ex [hunk 1]
```diff
@@ -21,7 +24,16 @@ defmodule Squadster.Accounts.Tasks.Notify do
   def notify([message: message, target: target, options: options]), do: message |> send_to(target, options)
 
   defp send_to(message, %User{} = user) do
-    mockable(HTTPoison).post @bot_endpoint, request_body(message, user |> Repo.preload(:settings)), @request_headers
+    user_with_settings = user |> Repo.preload(:settings)
+
+    if user_with_settings.settings.email_notifications_enabled and user.email do
```
Comment (ARtoriouSs) [inline lib/squadster/accounts/tasks/notify.ex:28]:
```elixir
%{settings: settings} = user |> Repo.preload(:settings)
if settings.email_notifications_enabled and user.email do
```

File: lib/squadster/mailer.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
+defmodule Squadster.Mailer do
+  @from "noreply@squadster.com"
```
Comment (ARtoriouSs) [inline lib/squadster/mailer.ex]: Can we set it in env variable too? Maybe FROM_EMAIL or smth?

File: lib/squadster/mailer/emails.ex [hunk 3]
```diff
@@ -0,0 +1,2 @@
+defmodule Blog.Mailer.Emails do
```
Comment (ARtoriouSs) [inline lib/squadster/mailer/emails.ex]: Blog? :)

File: mix.exs [hunk 4]
```diff
@@ -71,7 +71,8 @@ defmodule Squadster.MixProject do
       {:httpoison, "~> 1.7.0"},
       {:mockery, "~> 2.3.1", runtime: false},
       {:excoveralls, "~> 0.11.1"},
-
+      {:mailgun, "~> 0.1.2"},
+      {:poison, "~> 2.1", override: true},
```
Comment (ARtoriouSs) [inline mix.exs:75]: Why do we need this change?

**Review by ARtoriouSs — CHANGES_REQUESTED** [review]

**Comment by ARtoriouSs** [comment]
Also could you please add a test for the Notify module? Just mock mailer and ensure it's called when the user have setting and email, and vise-versa

File: lib/squadster/accounts/tasks/notify.ex (hunk 1 above)
Comment (str1nker) [reply lib/squadster/accounts/tasks/notify.ex:28]: the next function requires user with settings

File: lib/squadster/mailer.ex (hunk 2 above)
Comment (str1nker) [reply lib/squadster/mailer.ex]: yes we can

File: lib/squadster/mailer/emails.ex (hunk 3 above)
Comment (str1nker) [reply lib/squadster/mailer/emails.ex]: Oops

File: mix.exs (hunk 4 above)
Comment (str1nker) [reply mix.exs:75]: Because mailgun depends on poison 2.1

File: lib/squadster/accounts/tasks/notify.ex (hunk 1 above)
Comment (ARtoriouSs) [reply lib/squadster/accounts/tasks/notify.ex:28]: Ok

**Review by mirterious — APPROVED** [review]

**Review by ARtoriouSs — APPROVED** [review]

**Pull request merged**